core/src/main/java/io/confluent/rest/Application.java

```diff
@@ -199,6 +210,25 @@ protected void doStop() throws Exception {
     ServletContextHandler context = new ServletContextHandler(ServletContextHandler.SESSIONS);
     context.setContextPath("/");
     context.addServlet(servletHolder, "/*");
+    String authMethod = getConfiguration().getString(RestConfig.AUTHENTICATION_METHOD);
+    if (!"NONE".equals(authMethod)) {
```

Comment:
SSL auth is currently supported via `ssl.client.auth` and applied with this line of code: `sslContextFactory.setNeedClientAuth(config.getBoolean(RestConfig.SSL_CLIENT_AUTH_CONFIG));`. This could be modified to be a part of an auth abstraction.